studiopress/genesis-custom-blocks
# Make the `pre-commit` hook fix formatting

## Changes
* The pre-commit hook is now like Gutenberg's. It fixes formatting when possible and stages the fix before the commit. If it can't fix it, it blocks the commit.
* Removes the [husky](https://www.npmjs.com/package/husky) pre-commit hook package. It was outdated, so we either had to [migrate](https://github.com/typicode/husky-4-to-8) or remove it.
* So to avoid `husky`, this adds  the pre-commit hook with a `git` command in `package.json` `prepare`
* I'll still have to iterate on this. But maybe we could use this in other repos if it helps.

## Testing instructions
1. Do a fresh clone of this repo
2. `git checkout remove/husky`
3. `composer i && nvm use && npm i`
4. Create a JS formatting error. Maybe remove a space inside parentheses in a JS file.
5. Stage that change
6. `git commit -m "Should be fixed"`
7. Expected: That formatting error is fixed before being committed.
